frontend/src/InputForm.js

```diff
@@ -320,10 +321,59 @@ class InputForm extends React.Component {
   }
 
   submit() {
-    var json = JSON.stringify(this.state);
+    let submitState = JSON.parse(JSON.stringify(this.state));
+    submitState.courses = submitState.courses.map((course) => this.convertCourseSections(course));
+
+    submitState.criterion.timePreferences = submitState.criterion.timePreferences.map((times) => 
+        this.timeToTimeRange(0, times.startTime, times.endTime));
+    var json = JSON.stringify(submitState);
+    //! This will be changed to send the json to the backend
     console.log(json);
   } 
 
+  convertCourseSections(course) {
+    course['sections'] = course.sections.map((section) => {
+      let newSection = {professor: section['professor'], meetingTimes: []};
+      if (section.monday) {
+        newSection.meetingTimes.push(this.timeToTimeRange(1, section.startTime, section.endTime));
+      }
+      if (section.tuesday) {
+        newSection.meetingTimes.push(this.timeToTimeRange(2, section.startTime, section.endTime));
+      }
+      if (section.wednesday) {
+        newSection.meetingTimes.push(this.timeToTimeRange(3, section.startTime, section.endTime));
+      }
+      if (section.thursday) {
+        newSection.meetingTimes.push(this.timeToTimeRange(4, section.startTime, section.endTime));
+      }
+      if (section.friday) {
+        newSection.meetingTimes.push(this.timeToTimeRange(5, section.startTime, section.endTime));
+      }
```

Comment:
You might want to consider storing the section days differently to avoid this duplicate code. For example, it could be an array of integers:

Sections on Tuesday and Thursday could be stored as section.days = [2, 4]

You could then loop over section.days and do this for each day in the array:
`newSection.meetingTimes.push(this.timeToTimeRange(<current element in array>, section.startTime, section.endTime));`

Reply:
This would be ideal, but since the UI/user input states are one and the same, and the days of the week are checkboxes, the way that they're stored in state is as separate booleans. I'll check with Heidi to see if we can change the way that sections are stored, but for now this is the only option wrt: this function.